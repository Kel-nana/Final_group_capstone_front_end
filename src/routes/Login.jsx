import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import {
  AiOutlineHome,
  AiOutlineEye,
  AiOutlineEyeInvisible,
} from "react-icons/ai";
import axios from "axios";

function Login() {
  const [formData, setFormData] = useState({
    email: "",
    password: "",
  });

  const [showPassword, setShowPassword] = useState(false);
  const [showSuccessModal, setShowSuccessModal] = useState(false);
  const [message, setMessage] = useState("");

  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      // Create a JSON object from formData
      const jsonData = JSON.stringify({
        user: {
          email: formData.email,
          password: formData.password,
        },
      });

      // Make a POST request using Axios with the JSON data
      const response = await axios.post(
        "https://doctalk-r977.onrender.com/users/sign_in",
        jsonData,
        {
          headers: {
            "Content-Type": "application/json", // Set the content type to JSON
          },
        }
      );
      console.log(response);
      // Handle the response as needed
      if (response.status === 200) {
        setMessage("Login successfull");
        const token = response.headers.get("Authorization");
        // setMessage('Account created successfully'); // Set message to true
        // Clear the input field
        localStorage.setItem("token", token);
        setFormData({
          email: "",
          password: "",
        });
        setShowSuccessModal(true);
        // navigate("/doctors");
      } else {
        console.error("Authentication failed. Please try again");
      }
    } catch (error) {
      console.error("An error occurred:", error);
    }
  };

  const handleChange = (e) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value,
    });
  };
  const togglePasswordVisibility = () => {
    setShowPassword(!showPassword);
  };

  const closeSuccessModal = () => {
    setShowSuccessModal(false);
    navigate("/doctors");
  };

  return (
    <div>
      <div className="bg-slate-200">
        <div className="bg-grey-lighter min-h-screen flex flex-col">
          <div className="container max-w-sm mx-auto flex-1 flex flex-col items-center justify-center px-2">
            <div className="bg-white px-6 py-8 rounded shadow-md text-black w-full">
              <h1 className="mb-8 text-3xl text-center">Login</h1>
              <form onSubmit={handleSubmit}>
                <input
                  type="text"
                  className="block focus:outline-none border border-grey-light w-full p-3 rounded mb-4"
                  name="email"
                  value={formData.email}
                  onChange={handleChange}
                  placeholder="Email"
                />

                <div className="flex items-stretch justify-between border border-grey-light items-stretch w-full rounded mb-4">
                  <input
                    type={showPassword ? "text" : "password"}
                    className="focus:outline-none w-3/4 p-3"
                    name="password"
                    value={formData.password}
                    onChange={handleChange}
                    placeholder="Password"
                  />
                  <button type="button" onClick={togglePasswordVisibility}>
                    {showPassword ? (
                      <AiOutlineEyeInvisible className="w-6 h-6 mr-4" />
                    ) : (
                      <AiOutlineEye className="w-6 h-6 mr-4" />
                    )}
                  </button>
                </div>

                <div className="mb-6 flex items-center justify-between">
                  {/* Remember me checkbox */}
                  <div className="mb-[0.125rem] block min-h-[1.5rem]">
                    <input
                      id="checkbox1"
                      type="checkbox"
                      defaultValue=""
                      className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600 hover:cursor-pointer"
                    />{" "}
                    {/* eslint-disable jsx-a11y/label-has-associated-control */}
                    <label
                      className="inline-block pl-[0.15rem] hover:cursor-pointer"
                      htmlFor="checkbox1"
                    >
                      Remember me
                    </label>
                  </div>
                </div>

                <button
                  type="submit"
                  className="w-full text-center py-3 rounded bg-green-400 hover:bg-green-500 text-white py-2 px-4"
                >
                  Login
                </button>
              </form>
            </div>
            <div className="text-grey-dark mt-6">
              Don&apos;t have an account?{" "}
              <a
                className="no-underline border-b border-blue text-blue-500"
                href="../sign_up/"
              >
                Sign up
              </a>
            </div>

            <div className="text-grey-dark mt-6">
              <a
                className="no-underline border-b border-blue text-blue-500"
                href="/"
              >
                {" "}
                <AiOutlineHome className="w-6 h-6" />
              </a>
            </div>
          </div>
        </div>
      </div>
      {showSuccessModal && (
        <div className="fixed border border-green inset-0 flex items-center justify-center z-50">
          <div className="bg-green-300 p-6 rounded shadow-lg">
            <h2 className="text-2xl font-semibold mb-4">Login Successful</h2>
            <p>{message}</p>
            <div className="mt-4">
              <button
                className="bg-green-400 text-white px-4 py-2 rounded"
                onClick={closeSuccessModal}
              >
                OK
              </button>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

export default Login;
