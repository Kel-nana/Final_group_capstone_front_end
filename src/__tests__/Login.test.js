import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Login from '../routes/Login';

describe('Login component', () => {
  it('Shoul render the component correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
