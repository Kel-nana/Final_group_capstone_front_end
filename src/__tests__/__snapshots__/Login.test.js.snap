// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Shoul render the component correctly 1`] = `
<div>
  <div>
    <div
      class="bg-slate-200"
    >
      <div
        class="bg-grey-lighter min-h-screen flex flex-col"
      >
        <div
          class="container max-w-sm mx-auto flex-1 flex flex-col items-center justify-center px-2"
        >
          <div
            class="bg-white px-6 py-8 rounded shadow-md text-black w-full"
          >
            <h1
              class="mb-8 text-3xl text-center"
            >
              Login
            </h1>
            <form>
              <input
                class="block focus:outline-none border border-grey-light w-full p-3 rounded mb-4"
                name="email"
                placeholder="Email"
                type="text"
                value=""
              />
              <div
                class="flex items-stretch justify-between border border-grey-light items-stretch w-full rounded mb-4"
              >
                <input
                  class="focus:outline-none w-3/4 p-3"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <button
                  type="button"
                >
                  <svg
                    class="w-6 h-6 mr-4"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="mb-6 flex items-center justify-between"
              >
                <div
                  class="mb-[0.125rem] block min-h-[1.5rem]"
                >
                  <input
                    class="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600 hover:cursor-pointer"
                    id="checkbox1"
                    type="checkbox"
                    value=""
                  />
                   
                  <label
                    class="inline-block pl-[0.15rem] hover:cursor-pointer"
                    for="checkbox1"
                  >
                    Remember me
                  </label>
                </div>
              </div>
              <button
                class="w-full text-center py-3 rounded bg-green-400 hover:bg-green-500 text-white py-2 px-4"
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
          <div
            class="text-grey-dark mt-6"
          >
            Don't have an account?
             
            <a
              class="no-underline border-b border-blue text-blue-500"
              href="../sign_up/"
            >
              Sign up
            </a>
          </div>
          <div
            class="text-grey-dark mt-6"
          >
            <a
              class="no-underline border-b border-blue text-blue-500"
              href="/"
            >
               
              <svg
                class="w-6 h-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
